returns/result.py

```diff
@@ -603,3 +604,34 @@ def decorator(arg: _FirstType) -> Result[_NewValueType, _FirstType]:
             return Failure(arg)
 
     return decorator
+
+
+_AdditionalType = TypeVar('_AdditionalType')
+
+
+def partition(
+    containers: Iterable[
+        result.ResultBasedN[_ValueType, _ErrorType, _AdditionalType]
+    ],
+) -> tuple[List[_ValueType], List[_ErrorType]]:
+    """
+    Partition a list of results into successful and failed unwrapped values.
+
+    Preserves order.
+
+    .. code:: python
+
+        >>> from returns.result import Failure, Success, partition
+        >>> results = [Success(1), Failure(2), Success(3)]
+        >>> partition(results)
+        ([1, 3], [2])
+
+    """
+    successes = []
+    failures = []
+    for res in containers:
+        if isinstance(res, Success):
+            successes.append(res.unwrap())
+        else:
+            failures.append(res.failure())
```

Comment:
```suggestion
        try:
            successes.append(res.unwrap())
        except UnwrapFailedError:
            failures.append(res.failure())
```

Please, use `container` as a name, not `res`